arbnode/seq_coordinator.go

```diff
@@ -584,13 +588,34 @@ func (c *SeqCoordinator) Start(ctxIn context.Context) {
 }
 
 func (c *SeqCoordinator) StopAndWait() {
+	wasChosen := c.CurrentlyChosen()
+	c.StopWaiter.StopAndWait()
+	// normal context now closed, use parent context
+	ctx := c.StopWaiter.GetParentContext()
 	if c.CurrentlyChosen() {
-		_ = c.chosenOneRelease(c.GetContext())
+		wasChosen = true
 	}
 	if c.reportedAlive {
-		_ = c.livelinessRelease(c.GetContext())
+		_ = c.livelinessRelease(ctx)
+	}
+	if wasChosen {
+		_ = c.chosenOneRelease(ctx)
+		if c.config.SafeShutdownDelay != time.Duration(0) {
+			log.Info("Waiting for someone else to become main sequencer..")
+			var nextChosen string
+			for {
+				var err error
+				nextChosen, err = c.CurrentChosenSequencer(ctx)
+				if err == nil && nextChosen != "" && nextChosen != c.config.MyUrl() {
+					break
+				}
+				<-time.After(c.config.RetryInterval)
```

Comment:
We'll want a timeout here so the sequencer doesn't get hard killed / db corruption if it fails to switch the lockout